core/src/main/java/org/elasticsearch/search/suggest/SuggestionBuilder.java

```diff
@@ -197,7 +197,18 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
 
     protected abstract SuggestionBuilder<T> innerFromXContent(QueryParseContext parseContext, String name) throws IOException;
 
-    protected abstract SuggestionContext build(QueryShardContext context) throws IOException;
+    public SuggestionContext build(QueryShardContext context) throws IOException {
+        SuggestionContext suggestionContext = innerBuild(context);
+        // copy over common settings to each suggestion builder
+        populateCommonFields(context.getMapperService(), suggestionContext);
```

Comment:
@abeyad IMHO, the complexity of adding another abstract method just for pre-populating fields in the build method is not justifiable. I understand adding abstract `newInstance()` makes `SuggestionBuilder` cleaner :) but it also makes the sub-class implementations difficult to follow. 

I didn't see your second last reply before posting my previous comment :)